crates/telnet-host/tests/moot/suspend_notify.moot

```diff
@@ -0,0 +1,3 @@
+; suspend(1); notify(player, "200"); return 201;
+200
```

Comment:
The order of returned values is not obvious; seems like it's consistently this. I think that's OK?

Reply:
This is where having a custom `Session` that holds `notify` values vs the return value from `eval` would be important.

Reply:
I'm reading "custom" to imply you believe this test manages an explicit `Session` object, and you're saying that separate queues would be useful because we could separately assert. Apologies if I'm misreading that.

That's not the case: this test suite runs the daemon plus telnet-host, so the `Session` implementation is `RpcSession`.

But maybe that's not what you mean, and instead just mean that `RpcSession` should hold these values separately, which... doesn't seem useful in the "real world"? In terms of testing this means: we're not testing that `return` and `notify` don't somehow return each others values (which seems unlikely?), but instead we're testing that the order `notify` and `return` values are sent is deterministic (and what the order is), which does seem like a useful thing to test for.

Reply:
(I'm... not even sure if `notify()` works in the test suite in `kernel`; I put it here specifically to exercise the full e2e behavior of this complex weird thing)

Reply:
It definitely / probably wouldn't be useful as-is in the real world, esp over telnet. But the ability to distinguish between "told" notify values of one kind or another and tag or retrieve them separately would certainly be useful for a test like this that has potential to vary in behaviour depending on timing. 

I do worry about collecting 1 second pauses here and there in test runs, too.

In any case I think this is fine.